Makefile

```diff
@@ -7,32 +7,45 @@
 GOBIN = ./build/bin
 GO ?= latest
 GORUN = go run
+PROJECT_NAME = $(shell git remote get-url origin | xargs basename -s .git)
 
+#? geth: Build geth
 geth:
 	$(GORUN) build/ci.go install ./cmd/geth
 	@echo "Done building."
 	@echo "Run \"$(GOBIN)/geth\" to launch geth."
 
+#? all: Build all packages and executables, include abidump, abigen, bootnode, clef, devp2p, ethkey, evm, geth, p2psim, rlpdump
 all:
 	$(GORUN) build/ci.go install
 
+#? test: Run the tests
 test: all
 	$(GORUN) build/ci.go test
 
+#? lint: Run certain pre-selected linters
 lint: ## Run linters.
 	$(GORUN) build/ci.go lint
 
+#? clean: Clean go cache and built excutebles, and the auto generated folder
 clean:
 	go clean -cache
 	rm -fr build/_workspace/pkg/ $(GOBIN)/*
 
 # The devtools target installs tools required for 'go generate'.
 # You need to put $GOBIN (or $GOPATH/bin) in your PATH to use 'go generate'.
 
+#? devtools: Install pre-requirement developer tools
 devtools:
 	env GOBIN= go install golang.org/x/tools/cmd/stringer@latest
 	env GOBIN= go install github.com/fjl/gencodec@latest
 	env GOBIN= go install github.com/golang/protobuf/protoc-gen-go@latest
 	env GOBIN= go install ./cmd/abigen
 	@type "solc" 2> /dev/null || echo 'Please install solc'
 	@type "protoc" 2> /dev/null || echo 'Please install protoc'
```

Comment:
How about `Install recommended developer tools` ？

Reply:
The `devtools` target is for stuff that is required for `go generate` to work. We need `protoc` for generating the Trezor USB interface bindings.

Reply:
The target is actually a bit outdated because we track go-based tools differently now. They no longer need to be installed because we use `go run` in the `//go:generate` line. I'm also not sure we need the Solidity compiler anymore, because all the contract bindings have been removed.

Reply:
@holiman @fjl Hi sir, is there something else that needs to be added?

If I understand this correctly, we could try to remove the `devtools` target, and check if this will affect the new developer to initialize their development environment.